### ISSUE TITLE
Rule Change: Add self to global objects checked by no-implied-eval

### ISSUE THREAD

**Issue by sethamus** [eslint/eslint#19950]
### What rule do you want to change?

no-implied-eval

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
self.setTimeout("alert('hi')", 1000);
```

### What does the rule currently do for this code?

Currently, the rule does not flag the use of setTimeout when accessed via the self global object. Only window, global, and globalThis are checked.

### What will the rule do after it's changed?

After the change, the rule will also flag implied eval when setTimeout, setInterval, or similar methods are accessed via the self global object, in addition to the existing global objects. For example, the code above will be reported as a violation.

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by snitin315** [comment]
I think it should be covered by https://github.com/eslint/eslint/pull/19932

**Comment by fasttime** [comment]
I can see that this wasn't fixed by #19932 ([Playground Link](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWltcGxpZWQtZXZhbDogXCJlcnJvclwiICovXG4vKiBnbG9iYWwgc2VsZiAqL1xuXG5zZWxmLnNldFRpbWVvdXQoXCJhbGVydCgnaGknKVwiLCAxMDAwKTtcbiIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)).

`no-implied-eval` uses a predefined list of variables it considers the global object, and `self` is not included:

https://github.com/eslint/eslint/blob/f46fc6c137c951bc73cf3bd9446053c1b11f769b/lib/rules/no-implied-eval.js#L40-L44

Perhaps we could revisit this mechanism to allow configurable names as suggested in the discussion of #19804.

**Comment by sethamus** [comment]
@fasttime My understanding is you're suggesting we don't add self to the default list of global objects, but instead make the rule configurable. I think adding self to the default list would be the ideal solution, but if that's not possible, a configurable option would be a good alternative.

**Comment by JoshuaKGoldberg** [comment]
I'm in favor of `self`. As noted in https://github.com/eslint/eslint/issues/19923#issuecomment-3052987351, folks use ESLint for some very old codebases - including those that still use that alias of the global object.

**Comment by nzakas** [comment]
I think it makes sense to add `self`, which is an alias of `window` in browsers. It doesn't make sense to support one but not the other.

Marking as accepted.
